src/main/kotlin/it/pagopa/ecommerce/scheduler/queues/TransactionActivatedEventsConsumer.kt

```diff
@@ -59,19 +59,19 @@ class TransactionActivatedEventsConsumer(
 
     val transactionId = getTransactionIdFromPayload(BinaryData.fromBytes(payload))
     val baseTransaction = reduceEvents(transactionId, transactionsEventStoreRepository)
+    val refundable = baseTransaction.map { isTransactionRefundable(it) }
     val refundPipeline =
       baseTransaction
         .filter { transactionUtils.isTransientStatus(it.status) }
         .flatMap { tx ->
-          updateTransactionToExpired(
-            tx, transactionsExpiredEventStoreRepository, transactionsViewRepository)
-        }
-        .filter {
-          val refundable = isTransactionRefundable(it)
-          logger.info(
-            "Transaction ${it.transactionId.value} in status ${it.status}, refundable: $refundable")
-          refundable
+          refundable.flatMap { refundable ->
+            logger.info(
+              "Transaction ${tx.transactionId.value} in status ${tx.status}, refundable: $refundable")
+            updateTransactionToExpired(
+              tx, transactionsExpiredEventStoreRepository, transactionsViewRepository, refundable)
+          }
```

Comment:
Can it be simplified to:

```suggestion
        logger.info(
          "Transaction ${tx.transactionId.value} in status ${tx.status}, refundable: $refundable")
        updateTransactionToExpired(
          tx, transactionsExpiredEventStoreRepository, transactionsViewRepository, isTransactionRefundable(tx))
      }
```
?

Reply:
Yes it was my first implemetation, I changed to this just for avoid call isTransactionRefundable twice. I'll revert to the previous implemented version

Reply:
Fixed with https://github.com/pagopa/pagopa-ecommerce-scheduler-service/pull/38/commits/1bd9a40e971d1466cac32b1de1767c3bea96e013